### PR TITLE
 Implement support to create releases in mirror repositories

### PR DESCRIPTION
**Pull request by FiwareULPGC**
This PR implements the support for the functionality asked in #3:

* Now the webhook service is able to identify [**Release** events](https://developer.github.com/v3/activity/events/types/#releaseevent) and recreate a newly published release inside the mirror repository. 
* The setup function has been also updated to recreate all releases when a new mirror is created.
* A new backend utility has been added to recreate all the releases for the already existing mirror repositories.
* The documentation has been updated to reflect the changes in requirements and functionality.

### CODE REVIEW

File: script/README.md [hunk 1]
```diff
@@ -4,5 +4,7 @@
 * The **setup-mirror.py** script is provided to allow the initialization of a mirror repository without using the web service. It will also populate **conf.yaml** with the appropiate values and structure.
 * The **update-def-mirros.py** script allows the user to create or update the definition, name and the pull request bot URL in a list of repositories.
 * The **deny-pull-requests.py** script acts as the backend for the bot that automatically closes pull requests in mirrored repositories.
+* The **recreate-releases.py** script allows the user to delete and create every release in a list of repositories.
```
Comment (jmcanterafonseca) [inline script/README.md]: and create all the releases

File: script/deny-pull-requests.py [hunk 2]
```diff
@@ -12,61 +12,65 @@
...
-		body = json.loads(payload_or_payload_file)
-	except ValueError as error:
-		with open(payload_or_payload_file, "r") as f:
-			body = json.load(f)
 
-	directory = os.path.dirname(os.path.abspath(__file__))
-	with open(directory+"/conf.yaml", "r") as f:
-		config = yaml.load(f)
-		config["workspace"] = os.path.expanduser(config["workspace"])
+    try:
```
Comment (jmcanterafonseca) [inline script/deny-pull-requests.py]: is this change related to this PR?

File: script/deny-pull-requests.py [hunk 3]
```diff
@@ -12,61 +12,65 @@
...
-	g.close_pull_request(org_name, repo_name, pull_request_number, src_url)
+    g.close_pull_request(org_name, repo_name, pull_request_number, src_url)
 
 
 if __name__ == "__main__":
 
-	if len(sys.argv) < 3: 
-		print "Less than two arguments were passed."
-		sys.exit(-1)
+    if len(sys.argv) < 3:
```
Comment (jmcanterafonseca) [inline script/deny-pull-requests.py]: is this really needed or you could just check for sys.argv == 2 ?

File: script/deny-pull-requests.py (hunk 2 above)
Comment (FiwareULPGC) [inline script/deny-pull-requests.py]: Since we changed the source code and tested it again, we decided to make it PEP-8 compliant. This block appears as changed because we deleted a couple of lines above some duplicated empty lines and also enforced the 80 character line length.

File: script/README.md (hunk 1 above)
Comment (FiwareULPGC) [inline script/README.md]: The text is already changed in the new commit.

File: script/deny-pull-requests.py (hunk 2 above)
Comment (FiwareULPGC) [inline script/deny-pull-requests.py]: Since we changed the source code and tested it again, we decided to make it PEP-8 compliant. This block appears as changed because we deleted a couple of lines above some duplicated empty lines and also enforced the 80 character line length.

File: script/deny-pull-requests.py (hunk 3 above)
Comment (FiwareULPGC) [inline script/deny-pull-requests.py]: You are right, it's not needed like this. In the new commit we already changed it in every script file.

File: script/recreate-releases.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
+#! /usr/bin/env  python
```
Comment (jmcanterafonseca) [inline script/recreate-releases.py]: add comments to this module

File: script/update-def-mirrors.py [hunk 5]
```diff
@@ -11,27 +11,35 @@
 from utils.utils import *
```
Comment (jmcanterafonseca) [inline script/update-def-mirrors.py]: add comments to the module

**Comment by jmcanterafonseca** [comment]
LGTM, but before merging please add some comments at the beginning of the new modules and the existing.

File: script/setup-mirror.py [hunk 6]
```diff
@@ -1,5 +1,12 @@
 #! /usr/bin/env  python
 
+"""Script register a new repository to be mirrored and create the mirror.
```
Comment (FiwareULPGC) [inline script/setup-mirror.py]: Script register -> Script to register

File: script/update-def-mirrors.py [hunk 7]
```diff
@@ -1,5 +1,12 @@
 #! /usr/bin/env  python
 
+"""Script to update the mirror description and deny PR hook configuration.
+
+The operation is executed for every mirror repository defined in the
```
Comment (FiwareULPGC) [inline script/update-def-mirrors.py]: every mirror repository -> all repositories

**Comment by FiwareULPGC** [comment]
We have added comments to all Python modules in the new commit.
